packages/core/contracts/optimistic-assertor/implementation/examples/DataAsserter.sol

```diff
@@ -76,29 +71,23 @@ contract DataAsserter {
             assertionLiveness,
             defaultIdentifier
         );
-        assertionsData[dataAssertionId] = DataAssertion(dataId, data, asserter, false);
-        oaIdsToInternalIds[oaAssertionId] = dataAssertionId;
-        emit DataAsserted(dataId, data, asserter, oaAssertionId);
+        assertionsData[assertionId] = DataAssertion(dataId, data, asserter, false);
+        emit DataAsserted(dataId, data, asserter, assertionId);
     }
 
     // OptimisticAssertor callback.
     function assertionResolved(bytes32 assertionId, bool assertedTruthfully) public {
         require(msg.sender == address(oa));
         // If the assertion was true, then the data assertion is resolved.
         if (assertedTruthfully) {
-            assertionsData[oaIdsToInternalIds[assertionId]].resolved = true;
-            DataAssertion memory dataAssertion = assertionsData[oaIdsToInternalIds[assertionId]];
+            assertionsData[assertionId].resolved = true;
+            DataAssertion memory dataAssertion = assertionsData[assertionId];
             emit DataAssertionResolved(dataAssertion.dataId, dataAssertion.data, dataAssertion.asserter, assertionId);
             // Else delete the data assertion if it was false so the same asserter can assert it again.
```

Comment:
Should also fix the comment as now same asserter/dataId are no more blocked. The only reason for delete is to reclaim some gas.